internal/attacktechniques/azure/execution/vm-run-command/main.go

```diff
@@ -0,0 +1,99 @@
+package azure
+
+import (
+	"context"
+	_ "embed"
+	"time"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute"
+	"github.com/datadog/stratus-red-team/internal/providers"
+	"github.com/datadog/stratus-red-team/pkg/stratus"
+	"github.com/datadog/stratus-red-team/pkg/stratus/mitreattack"
+	"log"
+)
+
+//go:embed main.tf
+var tf []byte
+
+func init() {
+	stratus.GetRegistry().RegisterAttackTechnique(&stratus.AttackTechnique{
+		ID:           "azure.execution.vm-run-command",
+		FriendlyName: "Execute Commands on Virtual Machine using Run Command",
+		Description: `
+By utilizing the 'RunCommand' feature on a Virtual Machine, an attacker can pass:
+
+- Windows: PowerShell commands to the VM as SYSTEM.
+- Linux: Shell commands to the VM as root.
+
+References:
+
+- https://docs.microsoft.com/en-us/azure/virtual-machines/windows/run-command
+- https://docs.microsoft.com/en-us/azure/virtual-machines/linux/run-command
+- https://github.com/hausec/Azure-Attack-Matrix/blob/main/Execution/AZT201/AZT201-1.md
+
+Warm-up: 
+
+- Create a virtual machine
+
+Detonation: 
+
+- Invoke a RunCommand on the target virtual machine
+`,
+		Detection:                  "Identify `Microsoft.Compute/virtualMachines/runCommand/action` events in Azure Activity logs",
+		Platform:                   stratus.Azure,
+		IsIdempotent:               true,
+		MitreAttackTactics:         []mitreattack.Tactic{mitreattack.Execution},
+		PrerequisitesTerraformCode: tf,
+		Detonate:                   detonate,
+	})
+}
+
+func detonate(params map[string]string) error {
+	vmObjectId := params["vm_instance_object_id"]
+	vmName := params["vm_name"]
+	resourceGroup := params["resource_group_name"]
+
+	ctx := context.Background()
+	cred := providers.Azure().GetCredentials()
+	subscriptionID := providers.Azure().SubscriptionID
+	clientOptions := providers.Azure().ClientOptions
+
+	client, err := armcompute.NewVirtualMachineRunCommandsClient(subscriptionID, cred, clientOptions)
+	if err != nil {
+		log.Fatalf("failed to create client: %v", err)
+	}
+
+	log.Println("Issuing Run Command for VM instance " + vmObjectId)
+
+	poller, err := client.BeginCreateOrUpdate(ctx,
+		resourceGroup,
+		vmName,
+		"RunPowerShellScript",
+		armcompute.VirtualMachineRunCommand{
+			Location: to.Ptr("West US"),
```

Comment:
We shouldn't hardcode the region, ideally it should come from the environment. Let me look into it

Reply:
- [ ] find a way to not hardcode Azure locations